### PR TITLE
mgr/dashboard: fix duplicated rows when creating NFS export.

### PR DESCRIPTION
**Pull request by alfonsomthd**
- Show an error message if the same pool & namespace is used by more than 1 cluster.
- Fix error handling when no rgw daemons found.
- Update documentation:
  - You learn first that orchestrator-managed clusters are detected automatically (therefore the documentation that follows is exclusively for user-defined clusters).
  - Include nfs-ganesha in the security scope list.

Fixes: https://tracker.ceph.com/issues/50440
Signed-off-by: Alfonso Martínez <almartin@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/mgr/dashboard.rst [hunk 1]
```diff
@@ -1208,8 +1209,17 @@ A log entry may look like this::
...
-----------------------
+Support for NFS-Ganesha Clusters Deployed by the Orchestrator
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Ceph Dashboard can be used to manage NFS-Ganesha clusters deployed by the
+Orchestrator and will detect them automatically. For more details
+on deploying NFS-Ganesha clusters with the Orchestrator, please see :ref:`orchestrator-cli-stateless-services`.
+Or particularly, see :ref:`deploy-cephadm-nfs-ganesha` for how to deploy
+NFS-Ganesha clusters with the Cephadm backend.
+
```
Comment (varshar16) [inline doc/mgr/dashboard.rst:1220]: Add note for rook

**Comment by varshar16** [comment]
Dashboard does not detect conflict if the namespace is configured in this format`$ ceph dashboard set-ganesha-clusters-rados-pool-namespace <pool_name>[/<namespace>]` and nfs service is deployed by cephadm using same pool and namespace

**Comment by alfonsomthd** [comment]
> Dashboard does not detect conflict if the namespace is configured in this format`$ ceph dashboard set-ganesha-clusters-rados-pool-namespace <pool_name>[/<namespace>]` ...
@varshar16 Fixed.

File: doc/mgr/dashboard.rst (hunk 1 above)
Comment (alfonsomthd) [reply doc/mgr/dashboard.rst:1222]: @varshar16 Done.

**Comment by alfonsomthd** [comment]
jenkins test docs

**Comment by alfonsomthd** [comment]
jenkins test docs

**Comment by alfonsomthd** [comment]
jenkins render docs

**Comment by alfonsomthd** [comment]
jenkins render docs

**Review by varshar16 — APPROVED** [review]
The rook or dashboard doc needs to be updated about disabling rook mgr module when dashboard is enabled. Otherwise looks good.

File: src/pybind/mgr/dashboard/services/ganesha.py [hunk 2]
```diff
@@ -69,6 +66,18 @@ def _get_clusters_locations(cls):
...
 
+            # Check pool/namespace collision.
+            for clusters in [orch_result, result]:
+                for cluster_name, cluster_data in clusters.items():
+                    if cluster_data['pool'] == pool and cluster_data['namespace'] == namespace:
+                        orch_error_msg = ' that is deployed by the Orchestrator' \
+                            if cluster_data['type'] == ClusterType.ORCHESTRATOR else ''
+                        raise NFSException(
+                            f'Pool `{pool}` and namespace `{namespace}` are already in use by '
+                            f'NFS-Ganesha cluster called `{cluster_name}`{orch_error_msg}. '
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/services/ganesha.py:77]: I'd put the above string inline as well (something I like about f-strings is that you can really read the final message, so I'd put everything there:
```suggestion
                            f'NFS-Ganesha cluster called `{cluster_name}`{" that is deployed by the Orchestrator"  if cluster_data['type'] == ClusterType.ORCHESTRATOR else ''}. '
```

**Review by epuertat — APPROVED** [review]
Looks good to me! Thanks @alfonsomthd !

**Comment by avanthakkar** [comment]
jenkins test docs

**Comment by avanthakkar** [comment]
jenkins test dashboard

**Review by avanthakkar — APPROVED** [review]

File: src/pybind/mgr/dashboard/services/ganesha.py (hunk 2 above)
Comment (alfonsomthd) [reply src/pybind/mgr/dashboard/services/ganesha.py]: Done.

**Pull request merged**